File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.math.BigInteger;

class Solution {
    public int solution(int n, int k) {
        int answer = 0;

        for(String s : Integer.toString(n, k).split("0")) {
            if(s.isEmpty()) continue;
            BigInteger num = new BigInteger(s);
            if (num.isProbablePrime(10)) {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def convert(n, k):
    ref = ""

    while n != 0:
        ref = str(n % k) + ref
        n //= k

    return ref

def isPrime(num):
    if num == 1:
        return False

    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False

    return True

def solution(n, k):
    numString = convert(n, k)
    numArr = numString.split("0")

    answer = 0
    for num in numArr:
        if num != "":
            answer += (isPrime(int(num)))

    return answer